### PR TITLE
clarify privatelink can't connect across cloud providers

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr adds clarification that privatelink cannot connect across cloud providers. this was raised in [internal slack thread](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1729864420508889) and confirmed by support and engineers.

used a snippet to convey this across 4 pages.

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-cross-cloud-privatelink-dbt-labs.vercel.app/docs/cloud/secure/about-privatelink
- https://docs-getdbt-com-git-cross-cloud-privatelink-dbt-labs.vercel.app/docs/cloud/secure/databricks-privatelink
- https://docs-getdbt-com-git-cross-cloud-privatelink-dbt-labs.vercel.app/docs/cloud/secure/postgres-privatelink
- https://docs-getdbt-com-git-cross-cloud-privatelink-dbt-labs.vercel.app/docs/cloud/secure/redshift-privatelink
- https://docs-getdbt-com-git-cross-cloud-privatelink-dbt-labs.vercel.app/docs/cloud/secure/snowflake-privatelink

### CODE REVIEW

File: website/snippets/_privatelink-across-providers.md [hunk 1]
```diff
@@ -0,0 +1 @@
+PrivateLink endpoints can't connect across cloud providers. For a PrivateLink connection to work, both dbt Cloud and the server (like {props.type}) must be hosted on the same cloud provider. For example, AWS PrivateLink can’t connect to resources hosted on Azure, and Azure PrivateLink can’t connect to resources hosted on AWS.
```
Comment (mirnawong1) [inline website/snippets/_privatelink-across-providers.md]: do we want to mention what alternatives to consider if users _are_ in this situation? e.g. do they need to consider alternatives like a VPN or any other solution?

Comment (dhaworth) [reply website/snippets/_privatelink-across-providers.md]: I think adding additional options would just confuse people more. I think this is pretty good. I might tweak the last sentence a bit to something like: 

"For example, dbt Cloud hosted on AWS cannot connect via PrivateLink to services hosted on Azure, and dbt Cloud hosted on Azure can’t connect via Private Link to services hosted on AWS."

Small note that PrivateLink on AWS doesn't have a space, but on Azure it does (Private Link).

**Review by dhaworth — APPROVED** [review]

**Comment by awalther52** [comment]
Can you also change the first sentence in the opening paragraph of https://docs-getdbt-com-git-cross-cloud-privatelink-dbt-labs.vercel.app/docs/cloud/secure/about-privatelink? (quoted below). Maybe something more generic would be good, like just removing "AWS" the link to AWS 🤷 or get more specific by also calling out Azure. 

>PrivateLink enables a private connection from any dbt Cloud Multi-Tenant environment to your data platform hosted on AWS using [AWS PrivateLink](https://aws.amazon.com/privatelink/) technology.

File: website/docs/docs/cloud/secure/about-privatelink.md [hunk 2]
```diff
@@ -7,11 +7,14 @@ sidebar_label: "About PrivateLink"
 
 import SetUpPages from '/snippets/_available-tiers-privatelink.md';
 import PrivateLinkHostnameWarning from '/snippets/_privatelink-hostname-restriction.md';
+import CloudProviders from '/snippets/_privatelink-across-providers.md';
 
 <SetUpPages features={'/snippets/_available-tiers-privatelink.md'}/>
 
 PrivateLink enables a private connection from any dbt Cloud Multi-Tenant environment to your data platform hosted on AWS using [AWS PrivateLink](https://aws.amazon.com/privatelink/) technology. PrivateLink allows dbt Cloud customers to meet security and compliance controls as it allows connectivity between dbt Cloud and your data platform without traversing the public internet. This feature is supported in most regions across NA, Europe, and Asia, but [contact us](https://www.getdbt.com/contact/) if you have questions about availability.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/secure/about-privatelink.md]:
```suggestion
PrivateLink enables a private connection from any dbt Cloud Multi-Tenant environment to your data platform hosted on a cloud provider, such as [AWS](https://aws.amazon.com/privatelink/) or [Azure](https://azure.microsoft.com/en-us/products/private-link), using that provider’s PrivateLink technology. PrivateLink allows dbt Cloud customers to meet security and compliance controls as it allows connectivity between dbt Cloud and your data platform without traversing the public internet. This feature is supported in most regions across NA, Europe, and Asia, but [contact us](https://www.getdbt.com/contact/) if you have questions about availability.
```

File: website/snippets/_privatelink-across-providers.md (hunk 1 above)
Comment (mirnawong1) [inline website/snippets/_privatelink-across-providers.md]:
```suggestion
PrivateLink endpoints can't connect across cloud providers. For a PrivateLink connection to work, both dbt Cloud and the server (like {props.type}) must be hosted on the same cloud provider. For example, dbt Cloud hosted on AWS cannot connect via PrivateLink to services hosted on Azure, and dbt Cloud hosted on Azure can’t connect via Private Link to services hosted on AWS.
```

**Comment by mirnawong1** [comment]
great feedback @awalther52 @dhaworth, thank you! have folded in your feedback and will get this merged!

**Review by awalther52 — APPROVED** [review]
Awesome, thank you! I setup PLs on the dbt side, and when I read our docs I was confused about this cross cloud point. Now I think it's clear what we support and what we don't. Really appreciate the quick change here!

**Comment by mirnawong1** [comment]
> Awesome, thank you! I setup PLs on the dbt side, and when I read our docs I was confused about this cross cloud point. Now I think it's clear what we support and what we don't. Really appreciate the quick change here!

oh that's so cool and great to know! thank you for confirming and for your huge help!

**Pull request merged**